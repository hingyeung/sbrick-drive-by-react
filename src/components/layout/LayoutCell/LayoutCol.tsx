import * as React from 'react';
import * as classNames from 'classnames';
import { StatelessComponent } from 'react';

interface LayoutCellComponentProps {
  className?: string;
  componentClass?: any;
  sm?: number;
  md?: number;
  lg?: number;
  xl?: number;
}

const isValidBootstrapColumnValue = function(value: number) {
  return value >= 1 && value <= 12;
};

const buildBootstrapGridColumnClasses = function(props: any): any {
  return {
    [`col-sm-${props.sm}`]: props.sm && isValidBootstrapColumnValue(props.sm),
    [`col-md-${props.md}`]: props.md && isValidBootstrapColumnValue(props.md),
    [`col-lg-${props.lg}`]: props.lg && isValidBootstrapColumnValue(props.lg),
    [`col-xl-${props.xl}`]: props.xl && isValidBootstrapColumnValue(props.xl)
  };
};

const LayoutCol: StatelessComponent<LayoutCellComponentProps> = (props) => {
  const {
    componentClass: Component,
    className,
    ...otherProps
  } = props;

  return (
    <Component {...otherProps} className={classNames(className, 'col', buildBootstrapGridColumnClasses(otherProps))}/>
  );
};

LayoutCol.defaultProps = {
  componentClass: 'div'
};

export default LayoutCol;