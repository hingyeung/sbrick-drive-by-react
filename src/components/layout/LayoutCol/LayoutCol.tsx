import * as React from 'react';
import * as classNames from 'classnames';

interface LayoutColComponentProps {
  // className?: string;
  componentClass?: any;
  sm?: number;
  md?: number;
  lg?: number;
  xl?: number;
  propsForComponent?: object;
}

const isValidBootstrapColumnValue = function (value: number) {
  return value >= 1 && value <= 12;
};

const buildBootstrapGridColumnClasses = function (props: any): any {
  return {
    [`col-sm-${props.sm}`]: props.sm && isValidBootstrapColumnValue(props.sm),
    [`col-md-${props.md}`]: props.md && isValidBootstrapColumnValue(props.md),
    [`col-lg-${props.lg}`]: props.lg && isValidBootstrapColumnValue(props.lg),
    [`col-xl-${props.xl}`]: props.xl && isValidBootstrapColumnValue(props.xl)
  };
};

const LayoutCol: React.SFC<LayoutColComponentProps &
  React.HTMLAttributes<HTMLDivElement> &
  React.ClassAttributes<HTMLDivElement>> = (props) => {
  const {
    componentClass: Component,
    className,
    propsForComponent,
    ...otherProps,
  } = props;

  return (
    <Component
      {...propsForComponent}
      className={classNames(className, 'col', buildBootstrapGridColumnClasses(otherProps))}
      {...otherProps}
    />
  );
};

LayoutCol.defaultProps = {
  componentClass: 'div'
};

export default LayoutCol;