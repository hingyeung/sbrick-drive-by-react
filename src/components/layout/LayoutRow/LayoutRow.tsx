import * as React from 'react';
import * as classNames from 'classnames';

interface Props {
  className?: string;
  componentClass?: any;
}

const LayoutRow: React.SFC<Props> = (props) => {
  const {
    componentClass: Component,
    className,
    ...otherProps
  } = props;

  return (
    <Component {...otherProps} className={classNames(className, 'row')}/>
  );
};

LayoutRow.defaultProps = {
  componentClass: 'div'
};

export default LayoutRow;