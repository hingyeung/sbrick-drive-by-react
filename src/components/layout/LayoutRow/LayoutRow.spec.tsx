import LayoutRow from './LayoutRow';
import { shallow } from 'enzyme';
import * as React from 'react';
import LayoutCol from '../LayoutCell/LayoutCol';

describe('LayoutRow', () => {
  it('should render className in props', () => {
    const wrapper =
      shallow(<LayoutRow className="my-row"><LayoutCol className="my-cell">Cell</LayoutCol></LayoutRow>);

    expect(wrapper.find('.my-row')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render using specified componentClass in props', () => {
    const wrapper =
      shallow(
        <LayoutRow className="my-row" componentClass={'b'}>
          <LayoutCol className="my-cell">
            Col
          </LayoutCol>
        </LayoutRow>);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('b.my-row')).toHaveLength(1);
  });
});