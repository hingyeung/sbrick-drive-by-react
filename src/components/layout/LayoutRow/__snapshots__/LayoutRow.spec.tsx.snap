// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutRow should render className in props 1`] = `
<div
  className="my-row row"
>
  <LayoutCol
    className="my-cell"
    componentClass="div"
  >
    Cell
  </LayoutCol>
</div>
`;

exports[`LayoutRow should render using specified componentClass in props 1`] = `
<b
  className="my-row row"
>
  <LayoutCol
    className="my-cell"
    componentClass="div"
  >
    Col
  </LayoutCol>
</b>
`;
